python/lsst/meas/astrom/match_probabilistic_task.py

```diff
@@ -0,0 +1,174 @@
+        select_ref : `numpy.array`
+            A boolean array of the same length as `catalog_ref` selecting the sources that can be matched.
+        select_target : `numpy.array`
+            A boolean array of the same length as `catalog_target` selecting the sources that can be matched.
+        wcs : `lsst.afw.image.SkyWcs`
+            A coordinate system to convert catalog positions to sky coordinates. Only used if
+            `self.config.coords_ref_to_convert` is set.
+        logger : `logging.Logger`
+            A Logger for logging.
+        logging_n_rows : `int`
+            Number of matches to make before outputting incremental log message.
+
+        Returns
+        -------
+        catalog_out_ref : `pandas.DataFrame`
+            Reference matched catalog with indices of target matches.
+        catalog_out_target : `pandas.DataFrame`
+            Reference matched catalog with indices of target matches.
+        """
+        if logger is None:
+            logger = self.log
+
+        config = self.config
+
+        if config.column_order is None:
+            flux_tot = np.nansum(catalog_ref.loc[:, config.columns_ref_flux].values, axis=1)
+            catalog_ref['flux_total'] = flux_tot
+            if config.mag_brightest_ref != -np.inf or config.mag_faintest_ref != np.inf:
+                mag_tot = -2.5 * np.log10(flux_tot) + config.mag_zeropoint_ref
+                select_mag = (mag_tot >= config.mag_brightest_ref) & (
+                    mag_tot <= config.mag_faintest_ref)
+            else:
+                select_mag = np.isfinite(flux_tot)
+            if select_ref is None:
+                select_ref = select_mag
+            else:
+                select_ref &= select_mag
+
+        if config.coords_ref_to_convert:
+            ra_ref, dec_ref = [catalog_ref[column] for column in config.coords_ref_to_convert.keys()]
+            factor = config.coords_ref_factor
+            radec_true = [geom.SpherePoint(ra*factor, dec*factor, geom.degrees)
+                          for ra, dec in zip(ra_ref, dec_ref)]
+            xy_true = wcs.skyToPixel(radec_true)
+
+            for idx_coord, column_out in enumerate(config.coords_ref_to_convert.values()):
+                catalog_ref[column_out] = np.array([xy[idx_coord] for xy in xy_true])
+
+        select_additional = (len(config.columns_target_select_true)
+                             + len(config.columns_target_select_false)) > 0
+        if select_additional:
+            if select_target is None:
+                select_target = np.ones(len(catalog_target), dtype=bool)
+            for column in config.columns_target_select_true:
+                select_target &= catalog_target[column].values
+            for column in config.columns_target_select_false:
+                select_target &= ~catalog_target[column].values
+
+        logger.info(f'Beginning MatcherProbabilistic.match with {np.sum(select_ref)}/{len(select_ref)}'
```

Comment:
Please do not use f-strings with loggers. The logger must use `%` formatting and must be given as parameters at the end. The reason is to defer the stringification until we know we need it. There are four other places (at least) that need to be fixed.